File: luiz/ba3b.py
#! /usr/bin/env python


def genome_from_path(kmers):
    joined = [kmers[0]]
    for kmer in kmers[1:]:
        joined.append(kmer[-1])

    return "".join(joined)


assert genome_from_path("ACCGA CCGAA CGAAG GAAGC AAGCT".split()) == "ACCGAAGCT"


if __name__ == "__main__":
    with open("data/rosalind_ba3b.txt", "r") as dataset:
        text = [l.rstrip() for l in dataset.readlines()]
    print(genome_from_path(text))

File: luiz/src/bin/ba3b.rs
use std::env;
use std::error::Error;
use std::fs;

pub fn genome_from_path(text: &[&str]) -> String {
    let mut seq: String = text[0].into();
    for kmer in &text[1..] {
        seq.push(kmer.chars().last().unwrap());
    }
    seq
}

fn main() -> Result<(), Box<dyn Error>> {
    let input: String = env::args()
        .nth(1)
        .unwrap_or("data/rosalind_ba3b.txt".into());
    let data = fs::read_to_string(input)?;
    let lines: Vec<&str> = data.lines().collect();

    println!("{}", genome_from_path(&lines));

    Ok(())
}
